content/en/docs/guides/performance.md

```diff
@@ -0,0 +1,102 @@
+---
+title: Performance Best Practices
+description: A user guide of both general and language-specific best practices to improve performance.
+---
+
+### General
+
+*   Always **re-use stubs and channels** when possible.
+
+*   **Use keepalive pings** to keep HTTP/2 connections alive during periods of
+    inactivity to allow initial RPCs to be made quickly without a delay (i.e.
+    C++ channel arg GRPC_ARG_KEEPALIVE_TIME_MS). 
+
+*   **Use streaming RPCs** when wanting to avoid continuous RPC initiation,
+    which includes connection load balancing at the client-side, starting a new
+    HTTP/2 request at the transport layer, and invoking a user-defined method
+    handler on the server side. Streams are particularly well-suited to handling
+    a long-lived logical flow of data from the client-to-server,
+    server-to-client, or in both directions. Streams, however, cannot be load
+    balanced once they have started and can be hard to debug for stream
+    failures. \
+    ***Side note:*** *This does not apply to Python (see Python section for
+    details).*
+
+*   Each gRPC channel uses 0 or more HTTP/2 connections and each connection
+    usually has a limit on the number of concurrent streams. When the number of
+    active RPCs on the connection reaches this limit, additional RPCs are queued
+    in the client and must wait for active RPCs to finish before they are sent.
+    Applications with high load or long-lived streaming RPCs might see
+    performance issues because of this queueing. There are two possible
+    solutions:
+
+    1.  **Create a separate channel for each area of high load** in the
+        application, or
+
+    2.  **Use a pool of gRPC channels** to randomly distribute RPCs over
+        multiple connections (channels must have different channel args to
+        prevent re-use so define a use-specific channel arg such as channel
+        number). \
+        ***Side note:*** *The gRPC team has plans to add a feature to fix these
```

Comment:
The formatting isn't quite right here.  This note applies to this entire bullet, not just to the second work-around option.

Stepping back for a moment, though, I'm not sure it actually makes sense to cover this topic in a performance doc, since it's really more of a missing feature.  Most of the time when people run into this, it's not about performance, it's about things that just aren't working.  And I'd really rather not tell people to create multiple channels, even though we know that they sometimes need to do so today.

If we do have to mention this, let's at least point to https://github.com/grpc/grpc/issues/21386.

Reply:
@markdroth how about if this moves to an advanced topic or special topic section?

Reply:
... and of course with a caveat that the issue is a temporary problem, this is a temporary workaround, and that it won't be needed indefinitely in the future.

Reply:
I understand this not being so much a performance issue.. I'm fine with omitting it if you have a strong opinion @markdroth. Alternatively, I can make it an advanced topic with a link to grpc/grpc#21386. Let me know which you would prefer.

Reply:
I'm still not hearing an argument about how this has anything to do with performance, so I'm not really convinced it belongs here.  Although to be fair, my main concern is not actually about whether or not it belongs in this particular doc; it's that I'm not really comfortable recommending that people create multiple channels, regardless of which doc we do it in.  But I guess this is the only work-around for this problem that exists today, so documenting it is fair, regardless of how unhappy it makes me. :/

So I guess this is fine.  But please do add a link to grpc/grpc#21386 for more info.

Reply:
Added link to grpc/grpc#21386.